### PR TITLE
Add /delete_doc/ endpoint

### PR DESCRIPTION
**Pull request by DevElCuy**

### CODE REVIEW

File: doc_manager.py [hunk 1]
```diff
@@ -105,6 +110,32 @@ async def get_retrieved_nodes(self, memory: ContextualCompressionRetriever, func
...
+                must=[
+                    rest.FieldCondition(
+                        key="metadata.source_url",
+                        match=rest.MatchValue(value=source_url),
+                    )
+                ]
+            )
+            records, _ = self.client.scroll(collection_name=self.collection_name, scroll_filter=filter)
+            for record in records:
+                if (
```
Comment (sidhujag) [inline doc_manager.py]: you added the filter on source url why would the record not have the source url the same here? what is the point of checking on this if statement again?

File: doc_manager.py [hunk 2]
```diff
@@ -139,6 +170,25 @@ async def add_doc(self, function_input: DocAddInput):
...
         return "success", end - start
 
+    async def delete_doc(self, function_input: DocDeleteInput):
+        start = time.time()
+        if len(function_input.source_url) <= 0:
+            logging.warn("DocManager: Cannot delete document because data missing")
+            end = time.time()
+            return "fail", end - start
+        memory = self.load(function_input.api_key)
+        ids, _ = self.get_url_docs_ids(function_input.source_url)
```
Comment (sidhujag) [inline doc_manager.py]: you dont need to find IDs, you can delete via filter see prune_web in web manager for example

File: doc_manager.py (hunk 1 above)
Comment (DevElCuy) [reply doc_manager.py]: just making sure it is exact match

File: main.py [hunk 3]
```diff
@@ -127,6 +127,13 @@ async def addDoc(function_input: DocAddInput):
     results, elapsed_time = await doc_manager.add_doc(function_input)
     return {'response': results, 'elapsed_time': elapsed_time}
 
+@app.post('/delete_doc/')
+async def deleteDoc(function_input: DocDeleteInput):
+    """Endpoint to conduct delete HTML document from doc portal."""
+    logging.info('delete from Doc Portal')
+    results, elapsed_time = await doc_manager.delete_doc(function_input)
```
Comment (sidhujag) [inline main.py]: not sure if you need to await if no coroutine is being passed back to you

File: doc_manager.py [hunk 4]
```diff
@@ -31,6 +31,11 @@ class DocAddInput(BaseModel):
     html_doc: str
     category: str
 
+class DocDeleteInput(BaseModel):
```
Comment (sidhujag) [inline doc_manager.py:34]: do you need api_key and category then?

Comment (DevElCuy) [reply doc_manager.py:34]: Only the category. All docs from indexer have it set to "Alan.sys" at the moment

**Pull request merged**